### PR TITLE
Code cleanup around file system watching

### PR DESCRIPTION
**Pull request by lptr**
Followup to https://github.com/gradle/gradle/pull/18205

Some further refactors to clarify concepts around file system watching.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-grade test QF please.

**Comment by lptr** [comment]
@bot-gradle test QF please.

**Comment by lptr** [comment]
@bot-gradle test QF please.

**Comment by lptr** [comment]
@bot-gradle test this please.

File: subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherUpdater.java [hunk 1]
```diff
@@ -24,6 +25,51 @@
...
+ *
+ *     <dt>recently used watchable hierarchies</dt>
+ *     <dd>The list of file system hierarchies Gradle has accessed recently. These include build roots from recent builds
+ *     that we actually accessed.</dd>
+ *
+ *     <dt>watched files</dt>
+ *     <dd>A {@link org.gradle.internal.file.FileHierarchySet} of the files that we are currently watching.
+ *     This helps decide whether or not something is being watched in a quick way.</dd>
+ *
+ *     <dt>watched hierarchies</dt>
```
Comment (wolfs) [inline subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherUpdater.java:55]: We only need this for the hierarchical watchers. Then the doc here needs to change a little bit as well.

We said we'd move this to hierarchical watcher updater in a separate PR.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Pull request merged**